src/Data/TypeRepMap/Internal.hs

```diff
@@ -266,25 +264,53 @@ hoistWithKey f (TypeRepMap as bs ans ks) = TypeRepMap as bs newAns ks
     withTr t = withTypeable t f
 {-# INLINE hoistWithKey #-}
 
--- | The union of two 'TypeRepMap's using a combining function.
+-- | The union of two 'TypeRepMap's using a combining function for conflicting entries
 unionWith :: forall f. (forall x. Typeable x => f x -> f x -> f x) -> TypeRepMap f -> TypeRepMap f -> TypeRepMap f
-unionWith f m1 m2 = fromTriples
-                  $ toTripleList
-                  $ Map.unionWith combine
-                                  (fromTripleList $ toTriples m1)
-                                  (fromTripleList $ toTriples m2)
+unionWith f ma mb = do
+    fromSortedTriples $ mergeMaps orderingM1 orderingM2
```

Comment:
The result of "mergeMaps" *SHOULD* be sorted, but for some reason switch `fromSortedTriples` -> `fromTriples` does fix some of the failing tests... mergeMaps must be returning things in the wrong order somehow...